### PR TITLE
Update providers to latest kubevirtci

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
Kubevirt CI now has a 'rolling' OKD 4.1. So instead of using 4.1.x we just use 4.1 and it will provide whatever the latest version is.
OKD 4.1 provider is now aware of KUBEVIRT_NUM_NODES, it will create a number of worker nodes equal to KUBEVIRT_NUM_NODES - 1, with a minimum of 1.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by awels** [comment]
ci test please

File: cluster-up/cluster/okd-4.1.2/provider.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 set -e
 
 #This will get overwritten when run make generate, this will not be needed once https://github.com/kubevirt/kubevirtci/pull/126 is merged.
-image="okd-4.1@sha256:d452e8f910bd08b4aabe2a9b8fd82dc5984a3e95f7096b3ebd6c8ba836a5361d"
+image="okd-4.1@sha256:84ab0dcc301b1e5c801de4f30e9950641f65a39a8756663d02da6de70e22e19e"
```
Comment (mhenriks) [inline cluster-up/cluster/okd-4.1.2/provider.sh]: isn't `cluster-up` supposed to be "vendored" in?

Comment (awels) [reply cluster-up/cluster/okd-4.1.2/provider.sh]: yes, it is, but this is a test that the image I build for kubevirtci actually works. I am going to abandon this (or maybe merge, and once the kubevirtci PR is merged create a new PR that vendors that) if it works, and give the nod to kubevirtci that it works.

Comment (mhenriks) [reply cluster-up/cluster/okd-4.1.2/provider.sh]: okay may way to add a /hold for this then

Comment (awels) [reply cluster-up/cluster/okd-4.1.2/provider.sh]: Okay kubevirtci is updated, I will do a proper vendoring instead of this.

**Comment by awels** [comment]
ci test please

**Comment by awels** [comment]
ci test please

**Pull request merged**